app/models/supporter.rb

```diff
@@ -116,14 +116,73 @@ def full_address
     Format::Address.full_address(self.address, self.city, self.state_code)
   end
 
+  def address
+    self&.primary_address&.address
+  end
+
+  def address=(address)
+    if primary_address.present?
+      primary_address.update_attributes(address: address)
```

Comment:
I don't think this is same behavior as before. The issue is that `#update_attributes` (and `#update`) updates that attribute and then saves immediately. That's not really how attribute setters work in Rails. Normally attribute setters just change the value in memory and then are saved when a save happens.

Instead, I think this line should be:

```ruby
primary_address.address = address
```

You'll also have to make sure that any time Supporter is saved and the primary address has changed, that primary address is also saved.